internal/testutils/xds/e2e/server.go

```diff
@@ -175,8 +176,8 @@ func StartManagementServer(opts ManagementServerOptions) (*ManagementServer, err
 	// Start serving.
 	go gs.Serve(lis)
 	logger.Infof("xDS management server serving at: %v...", lis.Addr().String())
-
-	return mgmtServer, nil
+	t.Cleanup(mgmtServer.Stop)
```

Comment:
Super nits: 
I see ms.Stop() being called explicitly 7 times in other tests. Looking at the code, I think that should just be a no-op.  If we can remove those references, should we just unexport Stop()?

Reply:
Cleaned up other places calling `Stop` in a `defer` or `t.Cleanup`. But there is still one place where we want to stop the management server explicitly as part of the test. So, I'm not able to unexport `Stop` at this point.